src/Model/Report/Report.php

```diff
@@ -39,13 +39,16 @@ public function create($omit_send_mail = false): void
         parent::create();
 
         if (!$omit_send_mail) {
-            $mail = new ilMail($this->user_id);
-            $type = ['system'];
+            // This is bad practive to use the Init::init() here. But this is a ActiveRecord class which is
+            // already blown up a lot. Adding the NotificationSender as a dependency would overload this class even more.
+            // We should try to get rid of this Init::init() call in the future.
```

Comment:
I would suggest to add a TODO here as well!
```suggestion
            // TODO: We should try to get rid of this Init::init() call in the future.
```